### PR TITLE
 Fix Reyhan death trigger (fixes #4055)

### LINKED ISSUE #4055: Reyhan last of Abazan moving counters that don't exist.
Whenever a token creture dies on my side of the field Reyhan last of Abzans ability (Whenever a creature you control dies or is put into the command zone, if it had one or more +1/+1 counters on it, you may put that many +1/+1 counters on target creature.) triggers even when the token does not have any counters on it. After I select a creature to move the counters to, it places two counters on the creature that seemingly came from no were. 
Here is a video of the bug in action https://www.youtube.com/watch?v=qAzRbFWmnHw

### PR DESCRIPTION
**Pull request by SpikesCafe-google**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java [hunk 1]
```diff
@@ -108,19 +108,31 @@ public boolean checkEventType(GameEvent event, Game game) {
...
+        }
+        
+        Permanent permanent = (Permanent) game.getLastKnownInformation(event.getTargetId(), Zone.BATTLEFIELD);
+        // A creature you control
+        if (!permanent.getControllerId().equals(this.getControllerId()) || !permanent.isCreature()) {
+            return false;
+        }
+        
+        // If it had one or more +1/+1 counters on it
+        int countersOn = permanent.getCounters(game).getCount(CounterType.P1P1);
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java]: Permanent can be null -- need to check

File: Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java [hunk 2]
```diff
@@ -108,19 +108,31 @@ public boolean checkEventType(GameEvent event, Game game) {
...
+            return false;
         }
-        return false;
+        if (zcEvent.getToZone() != Zone.GRAVEYARD && zcEvent.getToZone() != Zone.COMMAND) {
+            return false;
+        }
+        
+        Permanent permanent = (Permanent) game.getLastKnownInformation(event.getTargetId(), Zone.BATTLEFIELD);
+        // A creature you control
+        if (!permanent.getControllerId().equals(this.getControllerId()) || !permanent.isCreature()) {
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java]: getControllerId() can be null -- need to check

File: Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java [hunk 3]
```diff
@@ -118,8 +118,9 @@ public boolean checkTrigger(GameEvent event, Game game) {
         }
         
         Permanent permanent = (Permanent) game.getLastKnownInformation(event.getTargetId(), Zone.BATTLEFIELD);
+        Player player = game.getPlayer(source.getControllerId());
         // A creature you control
-        if (!permanent.getControllerId().equals(this.getControllerId()) || !permanent.isCreature()) {
+        if (player == null || !player.equals(this.getControllerId()) || permanent == null || !permanent.isCreature()) {
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java]: `player.equals(this.getControllerId())` -- that' correct, but not now (player class do not compare by uuid -- see #4407). Need `player.getId().equals(this.getControllerId())`

File: Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java (hunk 1 above)
Comment (SpikesCafe-google) [reply Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java]: Done.

File: Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java (hunk 3 above)
Comment (SpikesCafe-google) [reply Mage.Sets/src/mage/cards/r/ReyhanLastOfTheAbzan.java]: Whoops, got it. Was too tired to make it happen last night.

**Pull request merged**